### PR TITLE
[PAL/Linux-SGX] Add explanation message on ECREATE's EIO error

### PR DESCRIPTION
**Pull request by dimakuv**
## Description of the changes 

In the last couple weeks, I had two independent debug sessions on this `-EIO` problem. It's always about the CPU feature bits mismatching in the SGX enclave's initial SECS and what exists on the platform.

Two examples:
1. People sometimes copy-paste the final `manifest.sgx` file from platform to platform and forget about `sgx.require_xxx` flags. They get this error and have no hint what this means.
2. There was/is a bug in QEMU v7.1 which was fixed in QEMU v8.0.1. If you're unlucky and have a bad QEMU version (which incorrectly propagates CPU feature bits), you get this EIO error.
   - Here I was wondering if I should add even more text to the message, mentioning that some versions of QEMU are buggy, and asking users to update to the very latest QEMU. But I think this is too corner-case to print such a message.

## How to test this PR? 

You can manually check with this diff:
```diff
diff --git a/CI-Examples/helloworld/helloworld.manifest.template b/CI-Examples/helloworld/helloworld.manifest.template
@@ -4,6 +4,8 @@ loader.entrypoint = "file:{{ gramine.libos }}"
 libos.entrypoint = "/helloworld"
 loader.log_level = "{{ log_level }}"

+sgx.require_amx = true
+
 loader.env.LD_LIBRARY_PATH = "/lib"

 fs.mounts = [
```

- Without this PR, you'll get:
   ```
   error: Enclave creation IOCTL failed: Input/output error (EIO)
   error: Creating enclave failed: Input/output error (EIO)
   error: load_enclave() failed with error: Input/output error (EIO)
   ```

- With this PR, you'll get:
   ```
   error: Enclave creation IOCTL failed with Input/output error (EIO). Probably your manifest requires CPU features (e.g. `sgx.require_avx512`) that are not available on this platform.
   error: Creating enclave failed: Input/output error (EIO)
   error: load_enclave() failed with error: Input/output error (EIO)
   ```

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1425)

### CODE REVIEW

**Review by dimakuv — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1425)* status: 0 of 1 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

___
*[`pal/src/host/linux-sgx/host_framework.c` line 533 at r1](https://reviewable.io/reviews/gramineproject/gramine/1425#-NYwo-pJCIzF-I6rbfzG:-NYwo-pJCIzF-I6rbfzH:b-331vqt) ([raw file](https://github.com/gramineproject/gramine/blob/7607fad2ee4a0525d2a871b9b10263ce6358df24/pal/src/host/linux-sgx/host_framework.c#L533)):*
> ```C
>     int ret = DO_SYSCALL(ioctl, g_isgx_device, SGX_IOC_ENCLAVE_INIT, &param);
>     if (ret < 0) {
>         log_error("Enclave initialization IOCTL failed: %s", unix_strerror(ret));
> ```

FYI: This hunk is purely for symmetry.
___
*[`pal/src/host/linux-sgx/host_framework.c` line 562 at r1](https://reviewable.io/reviews/gramineproject/gramine/1425#-NYwo2RB9cXZfT6JeVvH:-NYwo2RB9cXZfT6JeVvI:b-331vqt) ([raw file](https://github.com/gramineproject/gramine/blob/7607fad2ee4a0525d2a871b9b10263ce6358df24/pal/src/host/linux-sgx/host_framework.c#L562)):*
> ```C
>                 break;
>         }
>         log_error("Enclave initialization IOCTL failed: %s", error);
> ```

FYI: This hunk is purely for symmetry.

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1425)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

**Review by kailun-qin — APPROVED** [review]
> It's always about the CPU feature bits mismatching in the SGX enclave's initial SECS and what exists on the platform.

Not exactly. Note that SGX driver `SGX_IOC_ENCLAVE_CREATE` actually fails w/ `-EIO` in all ECREATE faults (elaborated in https://www.felixcloutier.com/x86/ecreate/Intel SDM), i.e., not only this reason of CPU feature mismatch. See https://github.com/torvalds/linux/blob/1ef6663a587ba3e57dc5065a477db1c64481eedd/arch/x86/kernel/cpu/sgx/ioctl.c#L100-L104 and https://github.com/torvalds/linux/blob/1ef6663a587ba3e57dc5065a477db1c64481eedd/arch/x86/kernel/cpu/sgx/encls.h#L95-L128 for details.

But I'm fine w/ the change.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1425#-:-NYxJKpQ7dwNQuvW1RAP:b-cxyxc4)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by kailun-qin — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/1425)* status: :shipit: complete! all files reviewed, all discussions resolved
